example/readme/Component.purs

```diff
@@ -0,0 +1,121 @@
+  ( name :: f Void     String String
+  , age  :: f AgeError String Age
+  ))
+
+derive instance newtypeDogForm :: Newtype (DogForm r f) _
+
+instance ToText AgeError where
+  toText = case _ of
+    InvalidInt -> "Age must be an integer"
+    TooLow -> "Age cannot be negative"
+    TooHigh -> "No dog has lived past 30 before"
+
+input :: forall m. Monad m => F.Input' DogForm m
+input =
+  { initialInputs: Nothing -- same as: Just (F.wrapInputFields { name: "", age: "" })
+  , validators: DogForm
+      { name: F.noValidation
+      , age: F.hoistFnE_ \str -> case Int.fromString str of
+          Nothing -> Left InvalidInt
+          Just n
+            | n < 0 -> Left TooLow
+            | n > 30 -> Left TooHigh
+            | otherwise -> Right (Age n)
+      }
+  }
+
+spec :: forall input m. Monad m => F.Spec' DogForm Dog input m
+spec = F.defaultSpec { render = render, handleEvent = F.raiseResult }
+  where
+  render st@{ form } =
+    UI.formContent_
+      [ HH.form
+        [ -- Using a form forces us to deal with an event. Using '\_ -> F.submit' here
+          -- would fire the event and cause the page to reload. Instead, we use
+          -- 'F.submitPreventDefault' to avoid firing the event unnecessarily
+          HE.onSubmit F.submitPreventDefault
+        ]
+        [ UI.input
+              { label: "Name"
+              , help: Right "Write your dog's name"
+              , placeholder: "Mila"
+              }
+              [ HP.value $ F.getInput _name st.form
+              , HE.onValueInput (F.setValidate _name)
+              ]
+        , UI.input
+              { label: "Age"
+              , help: UI.resultToHelp "Write your dog's age" $ F.getResult _age st.form
+              , placeholder: "3"
+              }
+              [ HP.value $ F.getInput _age form
+              , HE.onValueInput $ F.setValidate _age
+              ]
+        , UI.buttonPrimary
+            []
+            [ HH.text "Submit" ]
+        ]
+      ]
+    where
+    _name = Proxy :: Proxy "name"
+    _age = Proxy :: Proxy "age"
+
+data Action = HandleDogForm Dog
+
+component :: forall q i o m. MonadAff m => H.Component q i o m
+component = H.mkComponent
+  { initialState: const unit
+  , render: const render
+  , eval: H.mkEval $ H.defaultEval { handleAction = handleAction }
+  }
+  where
+  handleAction (HandleDogForm dog) = logShow (dog :: Dog)
+
+  render = 
+    UI.section_
+      [ UI.h1_ [ HH.text "Formless" ]
+      , UI.h2_ [ HH.text "A form using a default form element" ]
+      , UI.p_
+          """
+          In formless, if you use a default form you would notice a page refresh when submitting the form.
+          In the other examples, this is avoided by not using a form element. In this example, we use
+          a special function F.submitPreventDefault to avoid the page refresh.
+
+          You can download the examples and replace F.submitPreventDefault with
+          \_ -> F.submit to see the difference.
```

Comment:
The text explaining how `submitPreventDefault` changes the behavior of them form and how to modify the example is probably worth reviewing as well.

Reply:
I think this can be removed, probably, as you really ought to just use forms this way in general. I think it'd be better to add documentation to `submit` that says it can be used to submit the form imperatively, but that if you want to capture the form submission event and submit your form that you should use `submitPreventDefault`

Reply:
We introduced `submitPreventDefault` to avoid the breaking change. Maybe it could be changed to work the other way around in a future major release?

Reply:
Yea, I’d rather have ‘submit’ and ‘submit_’ using the usual Halogen naming conventions in the future!